### ISSUE TITLE
sourmash signature merge does not merge signatures

### ISSUE THREAD

**Issue by taylorreiter** [dib-lab/sourmash#668]
I generated signatures from paired-end data like this:

```
rule calc_sigs_raw:
    output: "outputs/sigs/raw/{sample}_raw.sig"
    input:
        R1="inputs/SRA/{sample}_1.fastq.gz",
        R2="inputs/SRA/{sample}_2.fastq.gz"
    shell:'''
    sourmash compute -k 21,31,51 --scaled 2000 --track-abundance -o {output} {input.R1} {input.R2}
    '''
```
When I went to run gather, I realized this created 6 signatures (3 k sizes for each of R1 and R2). I decided to run `sourmash merge` on the k31 pieces to get one signature to run gather with:

```
rule merge_sigs_raw:
    output: "outputs/sigs/raw/{sample}_raw_merged.sig"
    input: "outputs/sigs/raw/{sample}_raw.sig"
    shell:'''
    sourmash signature merge -k 31 -o {output} {input}
    '''
```

And this output a signature with no hashes. Did I run this wrong, or is this an issue with `sourmash merge`? Can also confirm that this occurs outside of snakemake. Using sourmash version 2.0.0.

**Comment by ctb** [comment]
```
sourmash compute -k 21,31,51 -o foo.sig ../GCF_000005845.2_ASM584v2_genomic.fna.gz ../GCF_000006945.1_ASM694v1_genomic.fna.gz
sourmash signature merge -k 31 -o xxx.sig foo.sig
sourmash signature describe xxx.sig
```
yields
```
== This is sourmash version 2.0.0a12.dev48+ga92289b. ==
== Please cite Brown and Irber (2016), doi:10.21105/joss.00027. ==

loaded 1 signatures total.
---
signature filename: xxx.sig
signature: bc1fde1b
source file: 
md5: bc1fde1bf83303696755781d9d7bd863
k=31 molecule=DNA num=500 scaled=0 seed=42 track_abundance=0
size: 500
signature license: CC0
```

so that seems at least not broken.

Trying with `--track-abundance --scaled 2000` triggers the bug, though:

```
sourmash compute -k 21,31,51 --scaled 2000 --track-abundance -o foo.sig ../GCF_000005845.2_ASM584v2_genomic.fna.gz ../GCF_000006945.1_ASM694v1_genomic.fna.gz
sourmash signature merge -k 31 -o xxx.sig foo.sig
sourmash signature describe xxx.sig
```

yields:

```
== This is sourmash version 2.0.0a12.dev48+ga92289b. ==
== Please cite Brown and Irber (2016), doi:10.21105/joss.00027. ==

loaded 1 signatures total.
---
signature filename: xxx.sig
signature: c16a5320
source file: 
md5: c16a5320fa475530d9583c34fd356ef5
k=31 molecule=DNA num=0 scaled=2000 seed=42 track_abundance=1
size: 0
signature license: CC0
```

**Comment by ctb** [comment]
Looks like `--track-abundance` is the problem, in the sense that `sourmash merge` is failing on signatures with `--track-abundance` set. When I use `--flatten` I see output.

**Comment by ctb** [comment]
Found it! This was a bug in the underlying `MinHash.merge` code. Fixed in #718.
